cores/esp8266/core_esp8266_waveform.cpp

```diff
@@ -173,137 +151,133 @@ int ICACHE_RAM_ATTR stopWaveform(uint8_t pin) {
-  uint32_t mask = 1<<pin;
-  if (!(waveformEnabled & mask)) {
+  uint32_t mask = 1UL<<pin;
+  if (!(waveformsEnabled & mask)) {
     return false; // It's not running, nothing to do here
   }
-  waveformToDisable |= mask;
-  // Ensure timely service....
-  if (T1L > microsecondsToClockCycles(10)) {
-    timer1_write(microsecondsToClockCycles(10));
-  }
-  while (waveformToDisable) {
+  waveformsToDisable |= mask;
+  timer1_write(microsecondsToClockCycles(2));
+  while (waveformsToDisable) {
     /* no-op */ // Can't delay() since stopWaveform may be called from an IRQ
   }
-  if (!waveformEnabled && !timer1CB) {
+  if (!waveformsEnabled && !timer1CB) {
     deinitTimer();
   }
   return true;
 }
 
-// The SDK and hardware take some time to actually get to our NMI code, so
-// decrement the next IRQ's timer value by a bit so we can actually catch the
-// real CPU cycle counter we want for the waveforms.
-#if F_CPU == 80000000
-  #define DELTAIRQ (microsecondsToClockCycles(3))
-#else
-  #define DELTAIRQ (microsecondsToClockCycles(2))
-#endif
-
-
 static ICACHE_RAM_ATTR void timer1Interrupt() {
   // Optimize the NMI inner loop by keeping track of the min and max GPIO that we
   // are generating.  In the common case (1 PWM) these may be the same pin and
   // we can avoid looking at the other pins.
   static int startPin = 0;
   static int endPin = 0;
 
-  uint32_t nextEventCycles = microsecondsToClockCycles(MAXIRQUS);
-  uint32_t timeoutCycle = GetCycleCountIRQ() + microsecondsToClockCycles(14);
+  constexpr uint32_t isrTimeoutCycles = microsecondsToClockCycles(14);
+  const uint32_t isrStart = ESP.getCycleCount();
+  uint32_t nextEventCycle = isrStart + microsecondsToClockCycles(MAXIRQUS);
 
-  if (waveformToEnable || waveformToDisable) {
+  if (waveformsToEnable || waveformsToDisable) {
     // Handle enable/disable requests from main app.
-    waveformEnabled = (waveformEnabled & ~waveformToDisable) | waveformToEnable; // Set the requested waveforms on/off
-    waveformState &= ~waveformToEnable;  // And clear the state of any just started
-    waveformToEnable = 0;
-    waveformToDisable = 0;
+    waveformsEnabled = (waveformsEnabled & ~waveformsToDisable) | waveformsToEnable; // Set the requested waveforms on/off
+    waveformsState &= ~waveformsToEnable;  // And clear the state of any just started
+    waveformsToEnable = 0;
+    waveformsToDisable = 0;
     // Find the first GPIO being generated by checking GCC's find-first-set (returns 1 + the bit of the first 1 in an int32_t)
-    startPin = __builtin_ffs(waveformEnabled) - 1;
+    startPin = __builtin_ffs(waveformsEnabled) - 1;
     // Find the last bit by subtracting off GCC's count-leading-zeros (no offset in this one)
-    endPin = 32 - __builtin_clz(waveformEnabled);
+    endPin = 32 - __builtin_clz(waveformsEnabled);
   }
 
   bool done = false;
-  if (waveformEnabled) {
+  if (waveformsEnabled) {
+    uint32_t now = ESP.getCycleCount();
```

Comment:
Moving the `now` out to here (once per IRQ call) seems like it is introducing an constant offset to the period of higher numbered GPIO pins vs. recalculating it as we consider each pin.  What's the thought process behind it?